autometacal/tests/test_model_fitting.py

```diff
@@ -0,0 +1,48 @@
+import tensorflow as tf
+
+import numpy as np
+from numpy.testing import assert_allclose
+
+import galflow as gf
+import galsim
+
+from autometacal.python import fitting
+
+# Some parameters used for testing shearing transforms
+N = 100
+gal_flux = 2.5    # counts
+gal_r0 = 1.7      # arcsec
+g1 = 0.2           #
+g2 = 0.3           #
+pixel_scale = 0.2  # arcsec / pixel
+
+def test_fitting():
+  """
+  This test generates a simple galaxy light profile with GalSim, shears it,
+  then estimate its flux and shape and checks that it recovers the right
+  flux and shapes (e1,e2).
+  """
+
+  gal = galsim.Exponential(flux=gal_flux, scale_radius=gal_r0)
+  # To make sure that GalSim is not cheating, i.e. using the analytic formula of the light profile
+  # when computing the affine transformation, it might be a good idea to instantiate the image as
+  # an interpolated image.
+  # We also make sure GalSim is using the same kind of interpolation as us (bilinear for TF)
+  reference_image = gal.drawImage(nx=N,ny=N, scale=pixel_scale)
+  gal = galsim.InterpolatedImage(reference_image,
+                                 x_interpolant='linear')
+
```

Comment:
In that case you don't need to go through the interpolation of the image and apply an affine transform. You can use the analytic formula which is closer to what happen in reality. Basically you can do:
```
gal = galsim.Exp()
gal = gal.shear()
image_galsim = gal.drawImage()
```
In that case you would use method='auto' for the drawing (which is the default).

Reply:
Alright, done!